### PR TITLE
tickets/DM-38797

### PR DESCRIPTION
**Pull request by gmegh**

### CODE REVIEW

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ShutDown"]
+
+import asyncio
+
+from lsst.ts import salobj
+
+
+class SetUp(salobj.BaseScript):
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:28]: Rename the Script `SetUp` -> `ShutDown`.

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        ------
+        asyncio.TimeoutError
+            If the Laser Tracker is not ready after the timeout.
+        """
+        await self.start_up()
+
+        await self.checkpoint("Shutting down Laser Tracker.")
+        await self.scheduler_remote.cmd_powerOff.set_start()
+        await self.checkpoint("Laser Tracker turned off.")
+
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:71]: I wonder if we should wait for the laser to report as OFF, like we do with the power on... I think it is probably a good idea.

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        """Run the script.
+        Raises
+        ------
+        asyncio.TimeoutError
+            If the Laser Tracker is not ready after the timeout.
+        """
+        await self.start_up()
+
+        await self.checkpoint("Shutting down Laser Tracker.")
+        await self.scheduler_remote.cmd_powerOff.set_start()
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:69]: This command will completely power off the laser tracker, later requiring someone to get inside the cell and powering it up again. I think we only want to power off the laser, which is done with the `laserPower` command but passing `power=0`, e.g.;

```
await self.laser_tracker.rem.lasertracker_1.cmd_laserPower.set_start(
             power=0,
             timeout=self.timeout_laser_power,
         )
```

**Review by tribeiro — COMMENTED** [review]
Please, make sure you reformat the code with black.

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py (hunk 1 above)
Comment (gmegh) [reply python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:28]: Done

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+    async def configure(self, config):
+        self.config = config
+
+    def set_metadata(self, metadata):
+        """Set estimated duration of the script."""
+
+        metadata.duration = self.timeout_laser_power + self.timeout_laser_warmdow
+
+    async def wait_laser_status_ok(self):
```
Comment (gmegh) [inline python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:72]: should i be calling this method from the set_up and passing OFF as attribute?

Comment (tribeiro) [reply python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:72]: You should probably rename the method `wait_laser_status_off`... 

I was thinking about this. We may want to actually have a `LaserTracker` class in ts_observatory_control and put all this functionality there. 

But I don't know.. For now I think we can keep these.

File: tests/test_maintel_lasertracker_shut_down.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            **{
+                "cmd_laserPower.set_start": self.mock_laser_power,
+                "evt_laserStatus.aget": self.get_laser_status,
+                "evt_laserStatus.next": self.get_laser_status,
+                "evt_summaryState.aget": self.get_summary_state,
+                "evt_summaryState.next": self.get_summary_state,
+            }
+        )
+
+        self.laser_status = types.SimpleNamespace(status=LaserStatus.OFF)
```
Comment (tribeiro) [inline tests/test_maintel_lasertracker_shut_down.py:50]: I think you want to start with `LaserStartus.ON`

**Review by tribeiro — COMMENTED** [review]
Please, remember to update the version history! 

I'm leaving a few inline comments but I think this is pretty close to being done. You can probably rebase/squash the commits into 1 single commit.

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py (hunk 4 above)
Comment (gmegh) [reply python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:72]: Done

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    async def run(self):
+        """Run the script."""
+
+        await self.checkpoint("Shutting down Laser Tracker.")
+        await self.laser_tracker.rem.lasertracker_1.cmd_laserPower.set_start(
+            power=0,
+            timeout=self.timeout_laser_power,
+        )
+
+        await self.checkpoint("Waiting for laser to warm down.")
```
Comment (tribeiro) [inline python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:99]: Consider something like:

```
"Waiting for laser to switch off."
```

It actually does not wait for it to cool down.

**Review by tribeiro — COMMENTED** [review]
Please, update the version history... I also have one final minor inline comment.

File: python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py (hunk 6 above)
Comment (gmegh) [reply python/lsst/ts/standardscripts/maintel/laser_tracker/shut_down.py:99]: Done

**Pull request merged**